armi/nuclearDataIO/rtflux.py

```diff
@@ -24,180 +41,172 @@
 import matplotlib.patches as mpatches
 from matplotlib.collections import PatchCollection
 
-from armi.nuclearDataIO import cccc
+from armi.nuclearDataIO import cccc, nuclearFileMetadata
 from armi.reactor import locations
 
 
-class RTFLUX(cccc.CCCCReader):
-    """
-    Read a binary RTFLUX or ATFLUX file from DIF3D output.
-
-    While NHFLUX stores nodal flux output data, RTFLUX stores finite difference flux output data.
-    While NHFLUX stores data by hexagon, RTFLUX stores data by triangle.
-    This can also read ATFLUX, which stores the finite difference adjoint flux output data.
-    """
-
-    def __init__(self, fName="RTFLUX"):
-        r"""
-        Initialize the RTFLUX or ATFLUX reader object.
-
-        Parameters
-        ----------
-        fName : str, optional
-            The file name of the RTFLUX/ATFLUX binary file to be read.
-
-        """
-        cccc.CCCCReader.__init__(self, fName)
+# See CCCC-IV documentation for definitions
+FILE_SPEC_1D_KEYS = (
+    "NDIM",
+    "NGROUP",
+    "NINTI",
+    "NINTJ",
+    "NINTK",
+    "ITER",
+    "EFFK",
+    "POWER",
+    "NBLOK",
+)
 
-        self.fc = {}  # file control info (sort of global for this library)
-        self.triangleFluxes = []
 
-        self.readFileID()  # Read RTFLUX file ID
-
-    def readAllData(self):
-        r"""
-        Read multigroup fluxes from the original DIF3D FD triangular-z mesh points.
-        RTFLUX contains the real fluxes, while ATFLUX contains the adjoint fluxes.
-
-        """
-
-        # Read basic data parameters (number of energy groups, assemblies, axial nodes, etc.)
-        self.read1D()
+class RtfluxData:
+    """
+    Data structure that is read from or written to a RTFLUX file.
+    """
 
-        # Read the hex ordering map between DIF3D "four color" nodal and DIF3D GEODST
-        # Also read index pointers to incoming partial currents on outer reactor surface
-        # (these don't belong to any assembly)
-        # Incoming partial currents are non-zero due to flux extrapolation
-        ng = self.fc["ngroup"]  # number of energy groups
-        imax = self.fc[
-            "ninti"
-        ]  # number of triangular mesh cells in "i" direction (rhombus or rectangle cells)
-        jmax = self.fc[
-            "nintj"
-        ]  # number of triangular mesh cells in "j" direction (rhombus or rectangle cells)
-        zmax = self.fc[
-            "nintk"
-        ]  # number of axial nodes (same for each assembly in DIF3D)
+    def __init__(self):
+        # Need Metadata subclass for default keys
+        self.metadata = nuclearFileMetadata._Metadata()
 
-        self.triangleFluxes = numpy.zeros((imax, jmax, zmax, ng))
+        self.groupFluxes = None
```

Comment:
What is this? An array of MG flux for a single region? A 2-D array by region/group?